code/sha3/Hacl.Impl.SHA3.fst

```diff
@@ -412,25 +433,63 @@ val absorb_last:
       as_seq h1 s ==
       S.absorb_last delimitedSuffix (v rateInBytes) (v rem)
         (as_seq h0 input) (as_seq h0 s))
-let absorb_last delimitedSuffix rateInBytes rem input s =
+let absorb_last_ delimitedSuffix rateInBytes rem input s =
   let h0 = ST.get() in
   let spec _ h1 =
     as_seq h1 s ==
     S.absorb_last delimitedSuffix (v rateInBytes) (v rem) (as_seq h0 input) (as_seq h0 s) /\
     live h1 s in
-  salloc1 h0 rateInBytes (u8 0) (Ghost.hide (loc s)) spec
-    (fun lastBlock ->
+  salloc1 h0 200ul (u8 0) (Ghost.hide (loc s)) spec
+    (fun temp ->
       let open Lib.RawIntTypes in
-       update_sub lastBlock (size 0) rem input;
+      let lastBlock = sub temp (size 0) rateInBytes in 
+      update_sub lastBlock (size 0) rem input;
       lastBlock.(rem) <- byte_to_uint8 delimitedSuffix;
       loadState rateInBytes lastBlock s;
       if not ((delimitedSuffix &. byte 0x80) =. byte 0) &&
          (size_to_UInt32 rem = size_to_UInt32 (rateInBytes -. 1ul))
       then state_permute s;
-      absorb_next s rateInBytes)
+      absorb_next s rateInBytes;
+      lemma_equiv_alloc_bigger_state #200 rateInBytes)
+
+val compare_more_zero_lte_b: a: size_t -> b: size_t -> Tot (r: bool {
```

Comment:
I'm not sure that introducing this auxiliary function helps. I'd rather inline the check.
If you prefer to keep it, it would need to be marked as `inline_for_extraction noextract` and the name could be improved (`positive_and_lte`?).